libs/s25main/desktops/dskCampaignSelection.cpp

```diff
@@ -0,0 +1,246 @@
+    const auto& selection = table->GetSelection();
+    boost::filesystem::path campaingPath = selection ? table->GetItemText(*selection, 5) : boost::filesystem::path();
+    if(campaingPath.empty())
+        throw std::runtime_error("Campaign path is empty.");
+    return campaingPath;
+}
+
+void dskCampaignSelection::Msg_ButtonClick(unsigned ctrl_id)
+{
+    if(ctrl_id == ID_Back)
+        WINDOWMANAGER.Switch(std::make_unique<dskSinglePlayer>());
+    else if(ctrl_id == ID_Next)
+        WINDOWMANAGER.Switch(std::make_unique<dskCampaignMissionSelection>(csi_, getSelectedCampaignPath()));
+}
+
+void dskCampaignSelection::Msg_Timer(unsigned ctrl_id)
+{
+    GetCtrl<ctrlTimer>(ctrl_id)->Stop();
+    FillCampaignsTable();
+}
+
+void dskCampaignSelection::FillCampaignsTable()
+{
+    const size_t numFaultyCampaignsPrior = brokenCampaignPaths_.size();
+    static const std::array<std::string, 2> campaignFolders = {
+      {s25::folders::campaignsBuiltin, s25::folders::campaignsUser}};
+
+    auto* table = GetCtrl<ctrlTable>(ID_Table);
+    for(const auto& campaignFolder : campaignFolders)
+    {
+        const bfs::path campaignPath = RTTRCONFIG.ExpandPath(campaignFolder);
+        for(const bfs::path& folder : ListDir(campaignPath, "", true))
+        {
+            if(helpers::contains(brokenCampaignPaths_, folder))
+                continue;
+
+            CampaignDescription desc;
+            CampaignDataLoader loader(desc, folder);
+            if(!loader.Load())
+            {
+                LOG.write(_("Failed to load campaign %1%.\n")) % folder;
+                brokenCampaignPaths_.insert(folder);
+                continue;
+            }
+
+            bool success = true;
+            for(auto const& mapName : desc.mapNames)
+            {
+                auto const mapPath = desc.getMapFilePath(mapName);
+                auto const luaFilepath = desc.getLuaFilePath(mapName);
+                if(!bfs::exists(mapPath))
+                {
+                    LOG.write(_("Campaign map %1% does not exist.\n")) % mapPath;
+                    brokenCampaignPaths_.insert(folder);
+                    success = false;
+                    continue;
+                }
+                if(!bfs::exists(luaFilepath))
+                {
+                    LOG.write(_("Campaign map lua file %1% does not exist.\n")) % luaFilepath;
+                    brokenCampaignPaths_.insert(folder);
+                    success = false;
+                    continue;
+                }
+
+                libsiedler2::Archiv map;
+                if(int ec = libsiedler2::loader::LoadMAP(mapPath, map, true))
+                {
+                    LOG.write(_("Failed to load map %1%: %2%\n")) % mapPath % libsiedler2::getErrorString(ec);
+                    brokenCampaignPaths_.insert(folder);
+                    success = false;
+                    continue;
+                }
+            }
+
+            if(success)
```

Comment:
This can never be false at this point, as you always `continue` where you set this to false, or am I missing anything?

Reply:
The continue effects only the inner loop over the maps. So if any error during check of the maps occours success will be false after the inner loop has finished. The continue is only there to avoid checking further stuff for an already failed. But i want to log all errors for all maps and not only the first failed map.